concourse/pipelines/container-build.jsonnet

```diff
@@ -0,0 +1,149 @@
+// Imports.
+local common = import '../templates/common.libsonnet';
+
+local buildcontainerimgtask = {
+  local task = self,
+
+  dockerfile:: 'Dockerfile',
+  input:: error 'must set input in buildcontainerimgtask',
+  context:: error 'must set context in buildcontainerimgtask',
+  destination:: error 'must set destination in buildcontainerimgtask',
+
+  platform: 'linux',
+  image_resource: {
+    type: 'registry-image',
+    source: {
+      repository: 'gcr.io/kaniko-project/executor',
+    },
+  },
+  inputs: [
+    { name: task.input },
+  ],
+  run: {
+    path: 'executor',
+    args: [
+      '--dockerfile=' + task.dockerfile,
+      '--context=' + task.context,
+      '--destination=' + task.destination,
+    ],
+  },
+};
+
+local buildcontainerimgjob = {
+  local job = self,
+
+  image:: error 'must set image in buildcontainerimgjob',
+  destination:: error 'must set destination in buildcontainerimgjob',
+  context:: error 'must set context in buildcontainerimgjob',
+  dockerfile:: 'Dockerfile',
+  input:: 'guest-test-infra',
+  passed:: '',
+  extra_steps:: [],
+  extra_resources:: [],
+
+  // Start of job definition
+  name: 'build-' + job.image,
+  serial_groups: ['serial'],
+  plan: [
+          {
+            get: job.input,
+            trigger: true,
+            [if job.passed != '' then 'passed']: [job.passed],
+          },
+        ] +
+        [
+          {
+            get: resource,
+            trigger: true,
+          }
+          for resource in job.extra_resources
+        ] +
+        job.extra_steps +
+        [
+
+          {
+            task: 'build-image',
+            config: buildcontainerimgtask {
+              destination: job.destination,
+              dockerfile: job.dockerfile,
+              context: job.context,
+              input: job.input,
+            },
+          },
+        ],
+};
+
+local BuildContainerImage(image) = buildcontainerimgjob {
+  repo:: 'gcr.io/gcp-guest',
+  image: image,
+  destination: '%s/%s:latest' % [self.repo, image],
+  context: 'guest-test-infra/container_images/' + image,
+};
+
+// Start of output.
+{
+  resources: [
+    common.GitResource('guest-test-infra'),
+    common.GitResource('compute-image-tools') {
+      source+: { paths: ['daisy_workflows/**'] },
+    },
+  ],
```

Comment:
Does this need the git resource for compute-daisy?

From the fly validate pipelines step on prow:

```
Error loading existing config:
  - invalid jobs:
	jobs.build-daisy.plan.do[0].get(compute-daisy): unknown resource 'compute-daisy'
```